pandas/core/nanops.py

```diff
@@ -421,7 +421,16 @@ def nanany(values, axis=None, skipna: bool = True, mask=None):
     >>> nanops.nanany(s)
     False
     """
-    values, _, _, _, _ = _get_values(values, skipna, fill_value=False, mask=mask)
+    values, _, dtype, _, _ = _get_values(values, skipna, fill_value=False, mask=mask)
+
+    # GH #12863
+    # Checking if the `axis` is None because numpy
+    # doesn't handle ``any`` and ``all`` on
+    # object arrays correclty. see
```

Comment:
correclty -> correctly.  small preference for this to go inside the "if", not sure how widely shared that is